### PR TITLE
Running with swiftlint

### PR DESCRIPTION
**Pull request by nas539**
I have just made some of the changes in the app after adding Swiftlint to get the app to still build, and have gotten rid of all the trailing white spaces

### CODE REVIEW

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 1]
```diff
@@ -45,15 +38,13 @@ class AddPlayerViewController: UITableViewController, UITextFieldDelegate, UICol
 			let player = Player(name: name, balance: balance, token: Token(rawValue: tokens[selectedToken])!)
 			BankManager.shared.players.append(player)
 			BankManager.shared.save()
-			let mainViewController = navigationController?.viewControllers.first as! MainViewController
-			mainViewController.playerCollectionView.reloadData()
-			mainViewController.playerNumberChanged()
+			let mainViewController = navigationController?.viewControllers.first as? MainViewController
+      mainViewController?.playerCollectionView.reloadData()
+      mainViewController!.playerNumberChanged()
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: Don't force unwrap

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 2]
```diff
@@ -63,52 +54,44 @@ class AddPlayerViewController: UITableViewController, UITextFieldDelegate, UICol
...
 	func numberOfSections(in collectionView: UICollectionView) -> Int {
 		return 1
 	}
-	
 	func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
 		return 12
 	}
-	
 	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    // swiftlint:disable force_cast
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: You can use this instead
`// swiftlint:disable:next force_cast`
Also, the comments should be lined up with the code.

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 3]
```diff
@@ -63,52 +54,44 @@ class AddPlayerViewController: UITableViewController, UITextFieldDelegate, UICol
...
 		let tokenName = isSelected ? tokens[indexPath.item] + "_filled" : tokens[indexPath.item]
 		tokenCell.tokenView.image = UIImage(named: tokenName)!.withRenderingMode(.alwaysTemplate)
 		let affineTransform = isSelected ? CGAffineTransform(scaleX: 1.1, y: 1.1) : CGAffineTransform.identity
 		UIView.animate(withDuration: 0.1, animations: {
 			tokenCell.transform = affineTransform
 		})
-		return tokenCell
-	}
-	
+		return tokenCell }
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: Curly brace should be on its own line, and there should be an empty line between functions

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 4]
```diff
@@ -63,52 +54,44 @@ class AddPlayerViewController: UITableViewController, UITextFieldDelegate, UICol
...
 	func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
 		selectedToken = indexPath.item
 		collectionView.reloadData()
 	}
-	
-	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: This looks like you changed the indentation for just this line. If you want the indentation to match the 2 spaces we have for our style guide, set it up for the whole file, not just a single line.

File: BoardBank/MonoBank/AddPlayerViewController.swift (hunk 3 above)
Comment (nas539) [reply BoardBank/MonoBank/AddPlayerViewController.swift]: Swiftlint sends warnings for trailing white spaces, thats why it look like that, but I will fix it! Thank you

File: BoardBank/MonoBank/MainViewController.swift [hunk 5]
```diff
@@ -276,41 +239,40 @@ class MainViewController: UIViewController, UICollectionViewDataSource, UICollec
...
 		let height = (playerCollectionView.bounds.height-verticalSpace-(playerCollectionView.bounds.height/8))/ceil(CGFloat(maxPlayers)/numberOfPlayersPerRow)
 
 		return CGSize(width: width, height: height)
 	}
-	
 	func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
 		if indexPath.item != 0 {
 			let cell = collectionView.cellForItem(at: indexPath)!
 			let player = BankManager.shared.players[indexPath.item-1]
+      // swiftlint:disable line_length
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MainViewController.swift]: For long lines like this, you can put each of the parameters on its own line. This should fix the issue.

File: BoardBank/MonoBank/SettingsViewController.swift [hunk 6]
```diff
@@ -63,7 +57,9 @@ class SettingsViewController: UITableViewController, UITextFieldDelegate {
 				BankManager.shared.soundsEnabled = soundsEnabledSwitch.isOn
 				BankManager.shared.save()
 				refreshText()
+        // swiftlint:disable force_cast
```
Comment (prrenfrew) [inline BoardBank/MonoBank/SettingsViewController.swift]: We don't need the force cast, correct? We could do an optional cast using `as?`

File: BoardBank/MonoBank/SettingsViewController.swift [hunk 7]
```diff
@@ -72,7 +68,9 @@ class SettingsViewController: UITableViewController, UITextFieldDelegate {
 				let resetAction = UIAlertAction(title: "New Game", style: .destructive, handler: { action in
 					BankManager.shared.players = [Player]()
 					BankManager.shared.save()
+          // swiftlint:disable force_cast
 					let mainViewController = self.navigationController?.viewControllers.first as! MainViewController
```
Comment (prrenfrew) [inline BoardBank/MonoBank/SettingsViewController.swift]: Same thing here. I think we can use `as?`

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 8]
```diff
@@ -45,15 +44,14 @@ class AddPlayerViewController: UITableViewController, UITextFieldDelegate, UICol
 			let player = Player(name: name, balance: balance, token: Token(rawValue: tokens[selectedToken])!)
 			BankManager.shared.players.append(player)
 			BankManager.shared.save()
-			let mainViewController = navigationController?.viewControllers.first as! MainViewController
-			mainViewController.playerCollectionView.reloadData()
-			mainViewController.playerNumberChanged()
+			let mainViewController = navigationController?.viewControllers.first as? MainViewController
+      mainViewController?.playerCollectionView.reloadData()
+      mainViewController?.playerNumberChanged()
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: Looks like the indentation is off.

File: BoardBank/MonoBank/BankManager.swift [hunk 9]
```diff
@@ -47,18 +41,18 @@ class BankManager {
 			self.quickAddAmount = quickAddAmount
 		}
 		if let playersData = UserDefaults.standard.object(forKey: "players") as? Data {
+      // swiftlint:disable:next force_cast
```
Comment (prrenfrew) [inline BoardBank/MonoBank/BankManager.swift]: Looks like indentation issues again. To quickly properly indent everything, select all the code using command+A, then re-indent with control+i.

**Comment by prrenfrew** [comment]
@nas539 fix conflicts.

**Comment by prrenfrew** [comment]
@nas539 this is currently giving me errors when I try to build it.

**Pull request merged**